### PR TITLE
Adding tests for hotplug with SCSI reservations

### PR DESCRIPTION
**Pull request by mkemel**
These tests cover parsing VM XML for Direct LUN hotplug with SCSI
Reservations, Drive creation and serializing to XML for libvirt

Related to https://github.com/oVirt/vdsm/pull/105

### CODE REVIEW

File: tests/virt/device_test.py [hunk 1]
```diff
@@ -536,6 +536,30 @@ class TestHotplug(TestCaseBase):
...
+      <driver name="qemu" io="native" type="raw" error_policy="stop"
+            cache="none"/>
+      <alias name="name1"/>
+      <address bus="0" controller="0" unit="1" type="drive" target="0"/>
+    </disk>
+  </devices>
+  <metadata xmlns:ovirt-vm="http://ovirt.org/vm/1.0">
+    <ovirt-vm:vm>
+      <ovirt-vm:device devtype="disk" name="sda">
+        <ovirt-vm:GUID>vm-guid</ovirt-vm:GUID>
```
Comment (bennyz) [inline tests/virt/device_test.py:557]: I'm pretty sure this is the WWID of the LUN

Comment (mkemel) [reply tests/virt/device_test.py:557]: Yeah, makes sense.

File: tests/virt/device_test.py [hunk 2]
```diff
@@ -536,6 +536,30 @@ class TestHotplug(TestCaseBase):
...
+      <driver name="qemu" io="native" type="raw" error_policy="stop"
+            cache="none"/>
+      <alias name="name1"/>
+      <address bus="0" controller="0" unit="1" type="drive" target="0"/>
+    </disk>
+  </devices>
+  <metadata xmlns:ovirt-vm="http://ovirt.org/vm/1.0">
+    <ovirt-vm:vm>
+      <ovirt-vm:device devtype="disk" name="sda">
+        <ovirt-vm:GUID>lun-uuid</ovirt-vm:GUID>
```
Comment (nirs) [inline tests/virt/device_test.py:557]: Based on the <source>, the value here should be "lun1".

**Review by nirs — COMMENTED** [review]
Looks good, minor issue with vm metadata. Please check with actual xml sent by engine.

File: tests/virt/device_test.py (hunk 2 above)
Comment (mkemel) [reply tests/virt/device_test.py:557]: Ack

**Review by nirs — APPROVED** [review]
Would be nice to get Milan ack but he can also look at it later when he comes backup.

**Comment by nirs** [comment]
@ahadas do you want to ack for Milan, or add someone else from the team to review the device tests?

**Comment by ahadas** [comment]
> @ahadas do you want to ack for Milan, or add someone else from the team to review the device tests?

@nyoxi can you please review?

**Review by nyoxi — APPROVED** [review]

**Pull request merged**